File: dex/field.py
from dex.access_flags import Field_AccessFlags
from dex.dex import Dex



def parse_access_flags(raw_access_flags):
    print(f"Starting aflag: {raw_access_flags}")
    parsed_access_flags = []
    for aflag in Field_AccessFlags:
        if raw_access_flags and isinstance(raw_access_flags, int):
            if aflag.value & raw_access_flags:
                parsed_access_flags.append(aflag)
                raw_access_flags -= aflag.value
                print(f"a_flags: {parsed_access_flags}, raw_flag: {raw_access_flags}")


class Field:
    def __init__(self, curr_idx: int, encoded_field: Dex.EncodedField, dex):
        self.encoded_field = encoded_field
        self.field_idx = curr_idx
        self.dex = dex
        self.fd = dex.fd

        self.access_flags = encoded_field.access_flags if type(encoded_field.access_flags) else parse_access_flags(encoded_field.access_flags)

        self.field_id_item: Dex.FieldIdItem = self.dex.dex.field_ids[self.field_idx]
        self.clazz_name = self.field_id_item.class_name
        self.type_name = self.field_id_item.type_name
        self.name = self.field_id_item.field_name

File: rust/platypus-dex/src/field.rs
/// Field representation — translates dex/field.py

use super::access_flags::{FieldAccessFlag, parse_field_access_flags};
use super::parser::{DexFileWithRaw, FieldIdItem};

#[derive(Debug, Clone)]
pub struct Field {
    pub field_idx: usize,
    pub access_flags: Vec<FieldAccessFlag>,
    pub class_name: String,
    pub type_name: String,
    pub name: String,
    /// Annotations attached to this field. Populated by Clazz::new
    /// from the owning ClassDefItem's `field_annotations` map.
    pub annotations: Vec<crate::parser::AnnotationItem>,
}

impl Field {
    pub fn new(curr_idx: usize, raw_access_flags: u64, dex: &DexFileWithRaw) -> Option<Self> {
        let field_id: &FieldIdItem = dex.parsed.field_ids.get(curr_idx)?;
        let access_flags = parse_field_access_flags(raw_access_flags as u32);

        Some(Field {
            field_idx: curr_idx,
            access_flags,
            class_name: field_id.class_name.clone(),
            type_name:  field_id.type_name.clone(),
            name:       field_id.field_name.clone(),
            // Annotations get layered on by Clazz::new — see the field
            // doc for the lookup model.
            annotations: Vec::new(),
        })
    }
}
